### ISSUE TITLE
cats.IO support

### ISSUE THREAD

**Issue by softshipper** [slick/slick#2126]
Hi Slick team 
I have a  question regarding to https://github.com/typelevel/cats-effect support. Do you plan in the future to support Cats.IO?  

Thanks

**Comment by hvesalai** [comment]
Hi @softshipper!

Slick is currently being maintained by its users who fix bugs and develop features that they need or want. As such, we don't have a roadmap or plans. If you want to start working on Cats.IO support, I'm sure you will find support for your efforts and even other developers who can help.

**Comment by pjfanning** [comment]
There are plenty of companion libraries that can be used to integrate Slick (Scala Futures based) with other effect systems like Cats-Effects and Zio. Examples include:
* https://github.com/kubukoz/slick-effect
* https://github.com/RMSone/slick-cats
* https://github.com/ScalaConsultants/zio-slick-interop

**Comment by nafg** [comment]
https://github.com/RMSone/slick-cats is forked under the `slick` organization.

Does anyone know who owns it though?

**Comment by nafg** [comment]
But it's not for cats IO. For cats IO use kubukoz/slick-effect

**Comment by hvesalai** [comment]
I would say the correct way to go would be to rip Futures away from Slick. I did analyze the thing a few years back and it wouldn't be such a big change.

**Comment by nafg** [comment]
👋 

That would be great, and is one of the things we can look into after Scala 3 support ships.

But what does your vision look like exactly?

**Comment by hvesalai** [comment]
I don't have a vision other than a complete rewrite of https://github.com/slick/slick/blob/main/slick/src/main/scala/slick/basic/BasicBackend.scala

Basically Slick needs asynchronous processing for three tings:
- waiting for a connection from a connection pool
- executing the query asynchronously
- mixing in user provided asynchronous actions

Currently we have a bad design when it comes to the connection pool management (as witnessed by the many issues posted regarding deadlocks, etc). Whether we stick with Future or go for cats.IO, we should make the connection pool use that abstraction as well.

**Comment by nafg** [comment]
There's a lot more Future usage than that file

**Comment by hvesalai** [comment]
Yes, but that file is the only place where Futures are being created, I think.

**Comment by nafg** [comment]
Sure, but it would be a much bigger change than that.

Do you want to have a chat about it?

**Comment by nafg** [comment]
@frosforever are you around?

**Comment by frosforever** [comment]
I'm not sure I have much say in this discussion. I'm no longer an active user/maintainer of slick or slick-cats.

**Comment by nafg** [comment]
@frosforever who owns slick-cats?

**Comment by hvesalai** [comment]
In any case, now there is Cats Effects for slick in my slick4 branch. Anybody interested, please have a spin. It's fully functional.

**Comment by hvesalai** [comment]
https://github.com/hvesalai/slick/tree/slick4
